crates/editor/src/editor.rs

```diff
@@ -914,62 +917,53 @@ impl CompletionsMenu {
                     .map(|(ix, mat)| {
                         let item_ix = start_ix + ix;
                         let candidate_id = mat.candidate_id;
-                        let completion = &completions_guard[candidate_id];
+                        let completion = completions_guard[candidate_id].clone();
 
                         let documentation = if show_completion_documentation {
                             &completion.documentation
                         } else {
                             &None
                         };
 
-                        let highlights = gpui::combine_highlights(
-                            mat.ranges().map(|range| (range, FontWeight::BOLD.into())),
-                            styled_runs_for_code_label(&completion.label, &style.syntax).map(
-                                |(range, mut highlight)| {
-                                    // Ignore font weight for syntax highlighting, as we'll use it
-                                    // for fuzzy matches.
-                                    highlight.font_weight = None;
-                                    (range, highlight)
-                                },
-                            ),
+                        let max_completion_len = px(510.);
```

Comment:
I'm not sure that this is always `510.`. When testing this on a javascript file I see:

<img width="395" alt="Screenshot 2024-03-06 at 20 23 39" src="https://github.com/zed-industries/zed/assets/94272/309cb10b-dd34-40cd-bd52-1bd362b60608">

Note that readline/promises is cut off but it shouldn't be, and the width is about 240px.

It would be ideal if we can pass this number down from the layout calculation so that they stay in sync.